### PR TITLE
feat: support Plugins Hub

### LINKED ISSUE #409: feature: support Plugins Hub
**Actual Behaviour**

APISIX has many plugins, but there is no list of plugins on the website.

**Expected Behaviour**

- Add a tab on the navbar to the home page.
- There is a list of all plugins Under the tab. 
- Click on any plugin, jump to the readme or get started tutorial.

**Screenshots of the issue**

 image

### PR DESCRIPTION
**Pull request by KishaniKandasamy**
Fixes: #409

Changes:
I have added a plugin hub by using list of plugins available in the APISIX dashboard.

Screenshots of the change:

https://user-images.githubusercontent.com/53715187/128417833-d1e298a8-7626-4d1d-adda-6c27f7047b4a.mp4

Please review when you are available

### CODE REVIEW

File: website/static/data/plugin.json [hunk 1]
```diff
@@ -0,0 +1,237 @@
...
+                "src": "static/img/plugin/defaultPluginImg.jpg"
+            },
+            {
+                "name": "ext-plugin-post-req",
+                "description": "The ext-plugin-post-req runs after executing builtin Lua plugins and before proxying to the upstream",
+                "src": "static/img/plugin/defaultPluginImg.jpg"
+            }
+        ]
+    }
+]
```
Comment (juzhiyuan) [inline website/static/data/plugin.json]:
```suggestion
]

```

File: website/static/data/plugin.json [hunk 2]
```diff
@@ -0,0 +1,237 @@
+[
```
Comment (juzhiyuan) [inline website/static/data/plugin.json:1]: Please format this file by using `2 spaces`

File: website/src/pages/pluginhub.js [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+        <PluginsContainer id={section.groupName}>
+          {pluginCards}
+        </PluginsContainer>
+      </div>
+    );
+  });
+
+  return (
+    <Page>
+      <PageTitle>APISIX®️ Plugin Hub</PageTitle>
```
Comment (juzhiyuan) [inline website/src/pages/pluginhub.js]:
```suggestion
      <PageTitle>Apache APISIX®️ Plugin Hub</PageTitle>
```

File: website/docusaurus.config.js [hunk 4]
```diff
@@ -462,9 +459,22 @@ module.exports = {
           position: "right",
         },
         {
-          to: "/help",
-          label: "Help",
+          label: "Resources",
           position: "right",
+          items: [
+            {
+              to: "/pluginhub",
```
Comment (juzhiyuan) [inline website/docusaurus.config.js]:
```suggestion
              to: "/hub",
```

File: website/static/data/plugin.json (hunk 2 above)
Comment (juzhiyuan) [reply website/static/data/plugin.json:1]: Maybe we could generate this file automatically in the future

**Review by juzhiyuan — COMMENTED** [review]
Thanks! It's very cool to see this page!

File: website/src/pages/pluginhub.js [hunk 5]
```diff
@@ -0,0 +1,138 @@
...
+
+
+function Plugins(props) {
+  const { siteConfig } = useDocusaurusContext();
+  const plugins = siteConfig.customFields.plugins.map((section) => {
+    const pluginCards = section.plugins.map((plugin) => {
+      return (
+        <div key={plugin.name}>
+          <PluginCard>
+            <a href={'https://apisix.apache.org/docs/apisix/plugins/' + plugin.name} target="_blank">
```
Comment (Baoyuantop) [inline website/src/pages/pluginhub.js]:
```suggestion
            <a href={`https://apisix.apache.org/docs/apisix/plugins/${plugin.name}`} target="_blank">
```

File: website/src/pages/pluginhub.js [hunk 6]
```diff
@@ -0,0 +1,138 @@
...
+              <img className="plugin-logo shadow" src={plugin.src} alt={plugin.name} />
+            </a>
+            <PluginName>{plugin.name}</PluginName>
+            <PluginDescription>{plugin.description}</PluginDescription>
+            <a className="read-more-link" href={'https://apisix.apache.org/docs/apisix/plugins/' + plugin.name}>{'Read more>>'}</a>
```
Comment (Baoyuantop) [inline website/src/pages/pluginhub.js]:
```suggestion
            <a className="read-more-link" href={`https://apisix.apache.org/docs/apisix/plugins/${plugin.name}`}>{'Read more>>'}</a>
```

File: website/static/data/plugin.json (hunk 2 above)
Comment (KishaniKandasamy) [reply website/static/data/plugin.json:1]: done

File: website/docusaurus.config.js [hunk 7]
```diff
@@ -462,9 +459,22 @@ module.exports = {
...
         },
         {
-          to: "/help",
-          label: "Help",
+          label: "Resources",
           position: "right",
+          items: [
+            {
+              to: "/hub",
+              label: "Pulgin Hub",
```
Comment (1502shivam-singh) [inline website/docusaurus.config.js]:
```suggestion
              label: "Plugin Hub",
```

File: website/src/pages/pluginhub.js [hunk 8]
```diff
@@ -0,0 +1,138 @@
...
+    const pluginCards = section.plugins.map((plugin) => {
+      return (
+        <div key={plugin.name}>
+          <PluginCard>
+            <a href={`https://apisix.apache.org/docs/apisix/plugins/${plugin.name}`} target="_blank">
+              <img className="plugin-logo shadow" src={plugin.src} alt={plugin.name} />
+            </a>
+            <PluginName>{plugin.name}</PluginName>
+            <PluginDescription>{plugin.description}</PluginDescription>
+            <a className="read-more-link" href={`https://apisix.apache.org/docs/apisix/plugins/${plugin.name}`}>{'Read more>>'}</a>
```
Comment (1502shivam-singh) [inline website/src/pages/pluginhub.js]:
```suggestion
            <a className="read-more-link" href={`https://apisix.apache.org/docs/apisix/plugins/${plugin.name}`}>{'Read more >'}</a>
```
I guess one `>` is enough, we have used this same style elsewhere also so. You may check the space between `Read more` and `>` according to your discretion to visual.

**Comment by 1502shivam-singh** [comment]
> Please review when you are available

Hey @KishaniKandasamy , really nice work :)
Something I would suggest, it's optional but would appreciate if it would be possible, you could add a fixed sidebar to this section (as in the wireframe I provided) to easily traverse between sections, you know to eliminate long scrolls. The labels in the sidebar can reference sections using anchor links so as to provide better usability. This is in view that this list will get longer eventually in the future.

**Comment by KishaniKandasamy** [comment]
> > Please review when you are available
> 
> Hey @KishaniKandasamy , really nice work :)
> Something I would suggest, it's optional but would appreciate if it would be possible, you could add a fixed sidebar to this section (as in the wireframe I provided) to easily traverse between sections, you know to eliminate long scrolls. The labels in the sidebar can reference sections using anchor links so as to provide better usability. This is in view that this list will get longer eventually in the future.

@1502shivam-singh  many thanks for your feedback ;) Yes I agree. I will commit those changes

**Comment by KishaniKandasamy** [comment]
Hi everyone , I have added a  sidebar .Please let me know your suggestions !

Screenshot (381)

**Comment by 1502shivam-singh** [comment]
> Hi everyone , I have added a sidebar .Please let me know your suggestions !
> 
> Screenshot (381)

Looks good. I guess you can keep the labels in sidebar as normal, non-italicised ones. Rest looks good to me.

**Comment by KishaniKandasamy** [comment]
> > Hi everyone , I have added a sidebar .Please let me know your suggestions !
> > Screenshot (381)
> 
> Looks good. I guess you can keep the labels in sidebar as normal, non-italicised ones. Rest looks good to me.

Done !

**Comment by 1502shivam-singh** [comment]
I guess we are clear then, rest LGTM.

**Review by 1502shivam-singh — APPROVED** [review]

File: website/src/pages/pluginhub.js [hunk 9]
```diff
@@ -0,0 +1,174 @@
...
+  margin-top: 84px;
+  text-transform: uppercase;
+`;
+
+
+function Plugins(props) {
+  const { siteConfig } = useDocusaurusContext();
+  const sidebar = siteConfig.customFields.plugins.map((section) => {
+    return (
+      <Sidebaritem><a className="sidebar-link" href={`#${section.groupName}`}>{section.groupName}</a></Sidebaritem>
```
Comment (juzhiyuan) [inline website/src/pages/pluginhub.js]:
```suggestion
      <Sidebaritem key={xxx}><a className="sidebar-link" href={`#${section.groupName}`}>{section.groupName}</a></Sidebaritem>
```

**Review by juzhiyuan — COMMENTED** [review]
Please take a look at the `map` key :)

**Comment by KishaniKandasamy** [comment]
> Please take a look at the `map` key :)
@juzhiyuan  fixed ;)

File: website/src/pages/pluginhub.js [hunk 10]
```diff
@@ -128,7 +128,7 @@ function Plugins(props) {
   const { siteConfig } = useDocusaurusContext();
   const sidebar = siteConfig.customFields.plugins.map((section) => {
     return (
-      <Sidebaritem><a className="sidebar-link" href={`#${section.groupName}`}>{section.groupName}</a></Sidebaritem>
+      <Sidebaritem key={xxx}><a className="sidebar-link" href={`#${section.groupName}`}>{section.groupName}</a></Sidebaritem>
```
Comment (juzhiyuan) [inline website/src/pages/pluginhub.js]: oh no, I mean there should have a unique key, see React Keys

Comment (KishaniKandasamy) [reply website/src/pages/pluginhub.js]: @juzhiyuan  I think I have already  changed 🤔 https://github.com/apache/apisix-website/pull/437/commits/89eddca70db5a152e93b5a314edf14b38935125a
s

**Review by juzhiyuan — APPROVED** [review]

**Comment by KishaniKandasamy** [comment]
> Fixes: #409
> 
> Changes:
> I have added a plugin hub by using list of plugins available in the APISIX dashboard.
> 
> Screenshots of the change:
> 
>  zoom_0.mp4 
> Please review when you are available

@Yiyiyimu    @liuxiran

**Review by liuxiran — APPROVED** [review]

**Comment by liuxiran** [comment]
merged, thanks for your contribution ~

**Pull request merged**